src/scaffold/tests.js

```diff
@@ -0,0 +1,185 @@
+  const eventsTypes = events.flatMap(event => event.inputs.map(input => ascTypeForEthereum(input.type))).filter(type => !type.startsWith("ethereum.") && !isNativeType(type))
+  const importTypes = [...new Set(eventsTypes)].join(', ')
+
+  return {
+    [`${strings.kebabCase(contract)}.test.ts`]: prettier.format(generateExampleTest(contract, event, indexEvents, importTypes), { parser: 'typescript', semi: false }),
+    [`${strings.kebabCase(contract)}-utils.ts`]: prettier.format(generateTestHelper(contract, events, importTypes), { parser: 'typescript', semi: false }),
+  }
+}
+
+/*
+  Generates the arguments that will be passed to the mock event function from the event inputs. Example:
+  let id = BigInt.fromI32(234)
+  let owner = Address.fromString("0x0000000000000000000000000000000000000001")
+  let displayName = "Example string value"
+  let imageUrl = "Example string value"
+*/
+const generateArguments = (eventInputs) => {
+  return eventInputs.map((input, index) => {
+    let ascType = ascTypeForEthereum(input.type)
+    return `let ${input.name || `param${index}`} = ${assignValue(ascType, input.name)}`
+  }).join('\n')
+}
+
+// Generates the value that will be assigned to a variable in generateArguments()
+const assignValue = (type) => {
+  switch (type) {
+    case "string":
+      return `"${VARIABLES_VALUES[type]}"`
+    case "BigInt":
+      return `BigInt.fromI32(${VARIABLES_VALUES[type]})`
+    case "Address":
+      return `Address.fromString("${VARIABLES_VALUES[type]}")`
+    case "Bytes":
+      return `Bytes.fromI32(${VARIABLES_VALUES[type]})`
+    case type.match(/Array<(.*?)>/)?.input:
+      innerType = type.match(/Array<(.*?)>/)[1]
+      return `[${assignValue(innerType)}]`
+    default:
+      let value = VARIABLES_VALUES[type]
+      return value ? value : `"${type} Not implemented"`
+  }
+}
+
+/*
+  Generates the assert.fieldEquals() for a given entity and event inputs. Example:
+  assert.fieldEquals(
+    "ExampleEntity",
+    "0xa16081f360e3847006db660bae1c6d1b2e17ec2a",
+    "owner",
+    "0x0000000000000000000000000000000000000001"
+  )
+*/
+const generateFieldsAssertions = (entity, eventInputs, indexEvents) => eventInputs.filter(input => input.name != "id").map((input, index) =>
+  `assert.fieldEquals(
+    "${entity}",
+    "0xa16081f360e3847006db660bae1c6d1b2e17ec2a${indexEvents ? "-1" : ""}",
+    "${input.name || `param${index}`}",
+    "${expectedValue(ascTypeForEthereum(input.type))}"
+  )`
+).join('\n')
+
+// Returns the expected value for a given type in generateFieldsAssertions()
+const expectedValue = type => {
+  switch (type) {
+    case type.match(/Array<(.*?)>/)?.input:
+      innerType = type.match(/Array<(.*?)>/)[1]
+      return `[${expectedValue(innerType)}]`
+    default:
+      let value = VARIABLES_VALUES[type]
+      return value ? value : `${type} Not implemented`
+  }
+}
+
+// Checks if the type is a native AS type or should be imported from graph-ts
+const isNativeType = type => {
```

Comment:
Should this function be considering all the types described [here](https://github.com/graphprotocol/graph-cli/blob/main/src/codegen/typescript.js#L106)? Maybe there's an overlap on what both do.

Reply:
Here i'm checking against the types after ethereum.Value to AS conversions https://github.com/graphprotocol/graph-cli/blob/main/src/codegen/types/conversions.js#L6 and i have to filter out the ones that are AS native and leave only the ones that get imported from graph-ts. In this case the only native AS types are boolean, Array<T>, i32 and string.

Reply:
But actually I just realised something i've missed, I don't handle arrays of graph-ts types, currently it will filter them. For example `Array<BigInt>` wi'll have to check the inner type too.

Reply:
Maybe this code generation should improve on the https://github.com/graphprotocol/graph-cli/blob/main/src/codegen/typescript.js code we have, so that it's easier to navigate and make assumptions about types.

But that can be improved in a future PR 🙂